src/mod_roster.erl

```diff
@@ -573,26 +582,30 @@ ask_to_pending(subscribe) -> out;
 ask_to_pending(unsubscribe) -> none;
 ask_to_pending(Ask) -> Ask.
 
--spec in_subscription(Acc :: mongoose_acc:t(),
-                      ToJID :: jid:jid(),
-                      FromJID :: jid:jid(),
-                      Type :: sub_presence(),
-                      Reason :: iodata()) ->
-    mongoose_acc:t().
-in_subscription(Acc, ToJID, FromJID, Type, Reason) ->
-    HostType = mongoose_acc:host_type(Acc),
+-spec in_subscription(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: mongoose_acc:t(),
+    Params :: #{to_jid := jid:jid(),
+                from_jid := jid:jid(),
+                type := sub_presence(),
+                reason := iodata()},
+    Extra :: gen_hook:extra().
+in_subscription(Acc,
+                #{to := ToJID, from := FromJID, type := Type, reason := Reason},
+                #{host_type := HostType}) ->
     Res = process_subscription(HostType, in, ToJID, FromJID, Type, Reason),
-    mongoose_acc:set(hook, result, Res, Acc).
-
--spec out_subscription(Acc :: mongoose_acc:t(),
-                       FromJID :: jid:jid(),
-                       ToJID :: jid:jid(),
-                       Type :: sub_presence()) ->
-    mongoose_acc:t().
-out_subscription(Acc, FromJID, ToJID, Type) ->
-    HostType = mongoose_acc:host_type(Acc),
+    {ok, mongoose_acc:set(hook, result, Res, Acc)}.
+
+-spec out_subscription(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: mongoose_acc:t(),
+    Params :: #{to_jid := jid:jid(),
+                from_jid := jid:jid(),
+                type := sub_presence()},
```

Comment:
Here the types are not those that are then used, dialyzer should fail on this one

Reply:
Thanks :) I forgot to change specs after applying Kamil's suggestion. I fixed it now.